gtda/images/filtrations.py

```diff
@@ -490,3 +490,143 @@ def transform(self, X, y=None):
         Xt = np.concatenate(Xt)
 
         return Xt
+
+
+@adapt_fit_transform_docs
+class ErosionFiltration(BaseEstimator, TransformerMixin):
+    """Filtrations of 2D/3D binary images based on the erosion of activated
+    regions.
+
+    This filtration assigns to each pixel in an image a grayscale value
+    calculated as follows. If the minimum Manhattan distance between the
+    pixel and any deactivated pixel in the image is less than or equal to
+    the parameter `n_iterations`, the assigned value is this distance –
+    in particular, deactivated pixels are assigned a value of 0.
+    Otherwise, the assigned grayscale value is the sum of the lengths
+    along all axes of the image – equivalently, it is the maximum
+    Manhattan distance between any two pixels in the image. The name of
+    this filtration comes from the fact that these values can be computed
+    by iteratively eroding activated regions, shrinking them by a total
+    amount `n_iterations`.
+
+    Parameters
+    ----------
+    n_iterations : int or None, optional, default: ``None``
+        Number of iterations in the erosion process. ``None`` means erosion
+        over the full image.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    n_iterations_ : int
+        Effective number of iterations in the erosion process. Set in
+        :meth:`fit`.
+
+    max_value_: float
+        Maximum pixel value among all pixels in all images of the collection.
+        Set in :meth:`fit`.
+
+    See also
+    --------
+    gtda.homology.CubicalPersistence, Binarizer
+
+    """
+    _hyperparameters = {'n_iterations_': [int, (1, np.inf)]}
+
+    def __init__(self, n_iterations=None, n_jobs=None):
+        self.n_iterations = n_iterations
+        self.n_jobs = n_jobs
+
+    def _calculate_erosion(self, X):
+        Xd = np.logical_not(X) * 1.
+
+        for iteration in range(1, min(self.n_iterations_,
+                                      self.max_value_) + 1):
+            Xtemp = np.asarray([ndi.binary_dilation(Xd[i])
+                                for i in range(Xd.shape[0])])
+            Xnew = (Xd + Xtemp) == 1
+            if np.any(Xnew):
+                Xd[Xnew] = iteration + 1
+            else:
+                break
+
+        mask_filtered = Xd == 0
+        Xd -= 1
+        Xd[mask_filtered] = self.max_value_
+        return Xd
```

Comment:
Looks incredibly similar to `_calculate_dilation`. Extract it?

Reply:
It is the dilation of the inverted image. I can extract it indeed.